### PR TITLE
Update ABL schema to consolidated structure

### PR DESCRIPTION
**Pull request by rnathuji**
This PR ties to the bakery changes in [this PR](https://github.com/openstax/output-producer-service/pull/319).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,37 +8,78 @@
...
+  }
+]
+```
+
+Archive Books:
+
+```json
+[
+  {
+      "repo": "col12067",
```
Comment (philschatz) [inline README.md:79]:
```suggestion
      "collection_id": "col12067",
```

Comment (philschatz) [inline README.md:79]:
```suggestion
      "collection_id": "col12067",
```

**Review by philschatz — CHANGES_REQUESTED** [review]
This PR looks good; I'm kicking myself for suggesting `repo` instead of `repository` or `repository_name` since all of the other fields are full words; what do you & CMs think?

File: README.md (hunk 1 above)
Comment (rnathuji) [reply README.md]: D'oh! Good 👁️  -- I fixed this.

**Comment by rnathuji** [comment]
> This PR looks good; I'm kicking myself for suggesting `repo` instead of `repository` or `repository_name` since all of the other fields are full words; what do you & CMs think?

🤔  - Yeah, now that you point it out I see the inconsistency. I'd lean towards changing to one of your suggestions unless CMs definitely want the shorter variant (maybe `repository_name` since it mirrors `collection_id` 🤷‍♂️ ).

**Comment by rnathuji** [comment]
@philschatz - This PR has been updated with the following:

* `repo` --> `repository_name` change
* Created the new ABL as `approved-book-list.json` with corresponding `abl-schema.json` so the existing `approved-books.json` and `schema.json` can stay where they were
* Placed `book-slugs.json` back in the repo for the time being just in case CMs find a need to make changes while still using the old `approved-books.json`
* Added a CI test that should fail if changes to `approved-books.json` or `book-slugs.json` are made without a corresponding updated to `approved-book-list.json`
* Added a `convert_abl.py` script to the repo with directions in the `README.md`

File: test.js [hunk 2]
```diff
@@ -89,3 +78,123 @@ test('all approved books have a slug defined in book-slugs.json', t => {
...
+  bookUUIDs.reduce((acc, entry) =>{
+    if (acc.includes(entry)) {
+      t.fail(`Duplicate book UUID found: ${entry}`)
+    }
+    acc.push(entry)
+    return acc
+  }, [])
+  t.pass()
+})
+
```
Comment (philschatz) [inline test.js:145]:
```suggestion

test('approved-book-list.json has at least one version for each book', t => {
  const listData = fs.readFileSync(approvedBookListPath, { encoding: 'utf8' })
  const abl = JSON.parse(listData)
  abl.approved_books.forEach(b => {
    if (b.repository_name) {
      t.truthy(abl.approved_versions.find(v => v.repository_name === b.repository_name), `No version for repository_name=${b.repository_name}`)
    } else {
      t.truthy(abl.approved_versions.find(v => v.collection_id === b.collection_id), `No version for collection_id=${b.collection_id}`)
    }
  })
})

```

It's unlikely to occur but since you added so many thorough tests here's one other corner case

**Review by philschatz — APPROVED** [review]
This looks great! And I think this does not impact REX or CMs workflow
